nomad/structs/structs.go

```diff
@@ -11069,6 +11069,48 @@ type AllocListStub struct {
 	ModifyTime            int64
 }
 
+// RescheduleEligible returns if the allocation is eligible to be rescheduled according
+// to its ReschedulePolicy and the current state of its reschedule trackers
+func (a *AllocListStub) RescheduleEligible(reschedulePolicy *ReschedulePolicy, failTime time.Time) bool {
```

Comment:
For now, this is a copy/paste from `Allocation.RescheduleEligible`. What approach you would recommend in order to prevent this code duplication?

Reply:
It looks like the only thing these methods reference from `Allocation` is the `RescheduleTracker`. Could we change it to be a method on `RescheduleTracker` itself? That's the same type between `Allocation` and `AllocListStub`.